app.js

```diff
@@ -6,13 +7,13 @@ const app = express();
 
 mongoose.Promise = require('bluebird');
 
-mongoose.connect('mongodb://test:test@ds257627.mlab.com:57627/first-blog-mern', { useMongoClient: true });
+mongoose.connect(`${process.env.DATABASE}://${process.env.DATABASE_USER}:${process.env.DATABASE_PASS}@${process.env.HOST}`, { useMongoClient: true });
```

Comment:
Your connect string will always start with `mongodb://`, so you don't need a variable there

Looks like you are missing the fields: `DATABASE_PORT` = 57627 and `DATABASE` = 'first-blog-mern'